.circleci/config.yml

```diff
@@ -0,0 +1,25 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/buildpack-deps:bionic
+        auth:
+          username: $DOCKERHUB_USERNAME
+          password: $DOCKERHUB_PASSWORD
+    steps:
+      - checkout
+      - run:
+          name: Clang-format test
+          command: |
+            set -o xtrace
+            curl -sSL "http://apt.llvm.org/llvm-snapshot.gpg.key" | sudo -E apt-key add -
+            echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-10 main" | sudo tee -a /etc/apt/sources.list > /dev/null
+            sudo -E apt-get -yq update >> ~/apt-get-update.log 2>&1
+            sudo -E apt-get -yq --no-install-suggests --no-install-recommends install clang-format-10
+            git diff -U0 --no-color HEAD^1 *.c *.cc *.cpp *.h *.hpp *.i *.ic *.ih | clang-format-diff-10 -style=file -p1 >_GIT_DIFF
```

Comment:
One doubt @marce, may be przemek knows, how do we deal when minimum required clang version changes?

Reply:
@satya-bodapati most likely we will have to update this script to reflect the changes.

Reply:
@satya-bodapati as per slack conversation. We will need to manually upgrade the circleci script to reflect the same version of .clang-format file. I will add a note into merge checklist ;)